connectors/sources/jira.py

```diff
@@ -0,0 +1,467 @@
+                retry += 1
+                if retry > self.retry_count:
+                    raise exception
+                logger.warning(
+                    f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}"
+                )
+                await asyncio.sleep(RETRY_INTERVAL**retry)
+
+    async def get_content(self, issue_key, attachment, timestamp=None, doit=False):
+        """Extracts the content for allowed file types.
+
+        Args:
+            issue_key (str): Issue key to generate `_id` for attachment document
+            attachment (dictionary): Formatted attachment document.
+            timestamp (timestamp, optional): Timestamp of attachment last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to False.
+
+        Returns:
+            dictionary: Content document with _id, _timestamp and attachment content
+        """
+        attachment_size = int(attachment["size"])
+        if not (self.enable_content_extraction and doit and attachment_size > 0):
+            return
+
+        attachment_name = attachment["filename"]
+        if os.path.splitext(attachment_name)[-1] not in TIKA_SUPPORTED_FILETYPES:
+            logger.debug(f"{attachment_name} can't be extracted")
+            return
+
+        if attachment_size > FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding the file content"
+            )
+            return
+
+        logger.debug(f"Downloading {attachment_name}")
+
+        document = {
+            "_id": f"{issue_key}-{attachment['id']}",
+            "_timestamp": attachment["created"],
+        }
+        temp_filename = ""
+        attachment_url = ATTACHMENT_CLOUD if self.is_cloud else ATTACHMENT_SERVER
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+            async for response in self._api_call(
+                url_name=attachment_url,
+                attachment_id=attachment["id"],
+                attachment_name=attachment["filename"],
+            ):
+                async for data in response.content.iter_chunked(CHUNK_SIZE):
+                    await async_buffer.write(data)
+                temp_filename = async_buffer.name
+
+        logger.debug(f"Calling convert_to_b64 for file : {attachment_name}")
+        await asyncio.to_thread(convert_to_b64, source=temp_filename)
+        async with aiofiles.open(file=temp_filename, mode="r") as async_buffer:
+            # base64 on macOS will add a EOL, so we strip() here
+            document["_attachment"] = (await async_buffer.read()).strip()
+        await remove(temp_filename)
+        return document
+
+    async def ping(self):
+        """Verify the connection with Jira"""
+        self._validate_configuration()
+        self.session = self._generate_session()
```

Comment:
Same as below - if it's used in `get_docs`, it should not get initialised in `ping`